domain_alias/domain_alias.module

```diff
@@ -64,3 +66,45 @@ function domain_alias_domain_operations(DomainInterface $domain, AccountInterfac
   }
   return $operations;
 }
+
+/**
+ * Implements hook_ENTITY_TYPE_load().
+ */
+function domain_alias_domain_load($entities) {
+  static $enabled;
+  // We can only perform meaningful actions if url.site is a cache context. Otherwise,
+  // the render process ignores our changes.
+  $required_cache_contexts = \Drupal::getContainer()->getParameter('renderer.config')['required_cache_contexts'];
+  if ($enabled === FALSE || (!in_array('url.site', $required_cache_contexts) && !in_array('url', $required_cache_contexts))) {
+    $enabled = FALSE;
```

Comment:
I dropped this in here for two reasons:

* To prevent running the code when the site hasn't been configured to leverage it. That's a performance gain, and the code should be further optimized for that, since we load the container multiple times. We should wrap this in an isset() check for $enabled.
* To make the config change a hard dependency, which makes support easier.

Perhaps I'm overthinking it, though.